h2o-algos/src/test/java/hex/adaboost/AdaBoostTest.java

```diff
@@ -538,4 +543,338 @@ public void testBasicTrainAndScoreDeepLearning() {
+    }
+
+    @Test
+    public void testBasicTrainAndScoreWeakLearnersEqualToDefaultGLM() {
+        try {
+            Scope.enter();
+            Frame train = parseTestFile("smalldata/prostate/prostate.csv");
+            String response = "CAPSULE";
+            int nlearners = 50;
+            train.toCategoricalCol(response);
+            Scope.track(train);
+            AdaBoostModel.AdaBoostParameters p = new AdaBoostModel.AdaBoostParameters();
+            p._train = train._key;
+            p._seed = 0xDECAF;
+            p._nlearners = nlearners;
+            p._weak_learner = AdaBoostModel.Algorithm.GLM;
+            p._response_column = response;
+
+            AdaBoost adaBoostReference = new AdaBoost(p);
+            AdaBoostModel adaBoostModelReference = adaBoostReference.trainModel().get();
+            Scope.track_generic(adaBoostModelReference);
+            assertNotNull(adaBoostModelReference);
+            Frame scoreReference = adaBoostModelReference.score(train);
+            Scope.track(scoreReference);
+
+            p._weak_learner_params = "{}";
+            AdaBoost adaBoost = new AdaBoost(p);
+            AdaBoostModel adaBoostModel = adaBoost.trainModel().get();
+            Scope.track_generic(adaBoostModel);
+            assertNotNull(adaBoostModel);
+
+            for (int i = 0; i < adaBoostModel._output.models.length; i++) {
+                System.out.println("GLM model = " + i);
+                GLMModel glmModel = DKV.getGet(adaBoostModel._output.models[i]);
+                assertEquals(50, glmModel._parms._max_iterations);
+            }
+
+            Frame score = adaBoostModel.score(train);
+            Scope.track(score);
+
+            assertFrameEquals(scoreReference, score, 0.0, 0.0);
+        } finally {
+            Scope.exit();
+        }
+    }
+
+    @Test
+    public void testBasicTrainAndScoreWeakLearnersEqualToDefaultGBM() {
+        try {
+            Scope.enter();
+            Frame train = parseTestFile("smalldata/prostate/prostate.csv");
+            String response = "CAPSULE";
+            int nlearners = 50;
+            train.toCategoricalCol(response);
+            Scope.track(train);
+            AdaBoostModel.AdaBoostParameters p = new AdaBoostModel.AdaBoostParameters();
+            p._train = train._key;
+            p._seed = 0xDECAF;
+            p._nlearners = nlearners;
+            p._weak_learner = AdaBoostModel.Algorithm.GBM;
+            p._response_column = response;
+
+            AdaBoost adaBoostReference = new AdaBoost(p);
+            AdaBoostModel adaBoostModelReference = adaBoostReference.trainModel().get();
+            Scope.track_generic(adaBoostModelReference);
+            assertNotNull(adaBoostModelReference);
+            Frame scoreReference = adaBoostModelReference.score(train);
+            Scope.track(scoreReference);
+
+            p._weak_learner_params = "{'ntrees':1, 'mtries':1, 'min_rows':1, 'sample_rate':1, 'max_depth':1, 'learn_rate':0.1}";
```

Comment:
How is it going look like from user prospective in python or R? Does user will have to specify `weak_learner_params ` as string? Couldn't we rather use algorithm objects as for StackedEnsembles?
E.g.:
```
week_learner_gbm = H2OGradientBoostingEstimator(distribution="bernoulli",
                                      ntrees=10,
                                      max_depth=3,
                                      min_rows=2,
                                      learn_rate=0.2,
                                      nfolds=nfolds,
                                      fold_assignment="Modulo",
                                      keep_cross_validation_predictions=True,
                                      seed=1)
adaboost_model = H2OAdaBoostEstimator(nlearners=50,
                                      learn_rate = 0.8,
                                      weak_learner = week_learner_gbm,
                                      seed=0xBEEF)
```

Reply:
Its going to be a python dictionary: https://github.com/h2oai/h2o-3/pull/15912/files#diff-380380d5155b7ac4f5cd9273929e9a5c893adb0eab3d149ca62053bbf451b463R29

Stacked Ensable deals with the different usecase -> it use already trained models and take the model_id from it and try to stack them together.

AdaBoost train models inside so we need only params to train them. By your suggestion we had to extract only parameters from the model and use them.

Reply:
In R it will be a list: https://github.com/h2oai/h2o-3/pull/15912/files#diff-49f529246b6d86bf2630442a92092d341f396a1f4e87124c5abad1d3c24bad10R14

The API is same for Infogram and AutoML. Only difference is that AutoML and Infogram expect list of e.g. `ntrees: [2]`. It doesn't make sence for AdaBoost so I expect only `ntrees:2`

Reply:
ok, sounds good to me